frontend/vite.config.mts

```diff
@@ -252,6 +253,8 @@ export default defineConfig({
       : {},
   },
   define: {
+    // Used when serving local assets in dev mode for islands.
+    "import.meta.env.VITE_MARIMO_ISLANDS": JSON.stringify(isIslands),
```

Comment:
nit: Vite automatically exposes `VITE_`-prefixed [env vars](https://vite.dev/guide/env-and-mode.html#env-variables) to `import.meta.env` as strings, so there's no need to manually inject `VITE_MARIMO_ISLANDS` via `define` config. 

Using `define` with `JSON.stringify(bool)` changes this behavior by inlining a boolean intead of a string, which can break type expectations (I believe TypeScript assumes `import.meta.env.VITE_*` values are string). 

We should update `isIslands()` from:

```ts
export function isIslands() {
  return import.meta.env.VITE_MARIMO_ISLANDS === true;
}
```

to:

```ts
export function isIslands() {
  return import.meta.env.VITE_MARIMO_ISLANDS === "true";
}
```